### PR TITLE
Provide a new way of coloring the weekview.

### PR DESCRIPTION
**Pull request by caske33**
Solves #48, solves #49

Provides 4 new background colors:
past and future (and optional past and future colors for the weekend, so they stand out). Today is rendered as partially past and partially future (based on the hour). There's also a line drawn at the current hour (on today's view).

(the new coloring scheme is optional, using the new `useNewColoringStyle` attribute)

### CODE REVIEW

**Comment by caske33** [comment]
(The reason I also changed mToday into a method is because while I was testing this, I started the app on a Tuesday and reopened it on a Wednesday. Without the method, the "sameDay" variable would be true for the Tuesday and false for the Wednesday. This is confusing (and wrong) behaviour and is therefore fixed in this version.

File: library/src/main/java/com/alamkanak/weekview/WeekView.java [hunk 1]
```diff
@@ -238,6 +248,13 @@ public WeekView(Context context, AttributeSet attrs, int defStyleAttr) {
             mHeaderRowPadding = a.getDimensionPixelSize(R.styleable.WeekView_headerRowPadding, mHeaderRowPadding);
             mHeaderRowBackgroundColor = a.getColor(R.styleable.WeekView_headerRowBackgroundColor, mHeaderRowBackgroundColor);
             mDayBackgroundColor = a.getColor(R.styleable.WeekView_dayBackgroundColor, mDayBackgroundColor);
+            mFutureBackgroundColor = a.getColor(R.styleable.WeekView_futureBackgroundColor, mFutureBackgroundColor);
```
Comment (rachitmishra) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: add new lines at end of the methods

Comment (caske33) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: ?

Comment (rachitmishra) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: while merging these will cause unwanted diffs!

**Comment by alamkanak** [comment]
@caske33 Since you are providing a way to let the weekends stand out, have you added any method to let users select the weekends? Or have you added the option to select the colors of each day individually? Because you know, the weekends are different in different countries. Currently the calendar does not have the concept of weekends.

**Comment by caske33** [comment]
Oh right, I didn't think about this. How would you suggest we implement this? With a callback method? (asking whether or not a certain day is weekend or not). Or is this possible in some way with android attributes?

**Comment by caske33** [comment]
@lolobosse @alamkanak How would you guys like to indicate the weekend? Is it ok if this is only possible through attributes and "bit flags"? 

Is there anything else that needs changing?

**Comment by lolobosse** [comment]
I think it should be ok if you do what you described.

**Comment by caske33** [comment]
merged by #182

**Pull request closed without merging**